sdk/main/include/Transaction.h

```diff
@@ -20,81 +20,145 @@
-  Transaction& operator=(const Transaction&) = default;
-  Transaction(const Transaction&&) = delete;
-  Transaction& operator=(const Transaction&&) = delete;
+  /**
+   * Set the valid transaction duration.
+   *
+   * @param duration The duration to set.
+   * @return Reference to this derived Transaction object.
+   */
+  SdkRequestType& setValidTransactionDuration(const std::chrono::duration<double>& duration);
+
+  /**
+   * Set the max transaction fee.
+   *
+   * @param fee The max transaction fee to set.
+   * @return Reference to this derived Transaction object.
+   */
+  SdkRequestType& setMaxTransactionFee(const Hbar& fee);
+
+  /**
+   * Set the transaction memo.
+   *
+   * @param memo The transaction memo to set.
+   * @return Reference to this derived Transaction object.
+   */
+  SdkRequestType& setTransactionMemo(const std::string& memo);
+
+  /**
+   * Set the transaction ID.
+   *
+   * @param id The transaction ID to set.
+   * @return Reference to this derived Transaction object.
+   */
+  SdkRequestType& setTransactionId(const TransactionId& id);
+
+  /**
+   * Extract the valid transaction duration.
+   *
+   * @return The valid transaction duration.
+   */
+  inline std::chrono::duration<double> getValidTransactionDuration() const { return mTransactionValidDuration; }
 
-  Transaction(
-    const std::unordered_map<
-      TransactionId,
-      std::unordered_map<AccountId, proto::TransactionBody>>& transactions)
-  {
-    (void)transactions;
-  }
+  /**
+   * Extract the max transaction fee.
+   *
+   * @return The max transaction fee.
+   */
+  inline Hbar getMaxTransactionFee() const { return mMaxTransactionFee; }
 
-  Transaction(const proto::TransactionBody& transaction) { (void)transaction; }
+  /**
+   * Extract the default max transaction fee.
+   *
+   * @return The default max transaction fee.
+   */
+  inline Hbar getDefaultMaxTransactionFee() const { return Hbar(2LL); }
+
+  /**
+   * Extract the transaction memo.
+   *
+   * @return The transaction memo.
+   */
+  inline std::string getTransactionMemo() const { return mTransactionMemo; }
 
   /**
-   * Validate the checksums.
+   * Extract the transaction ID.
    *
-   * @param client The client with which to validate the checksums
+   * @return The transaction ID.
    */
-  virtual void validateChecksums(const Client& client) const = 0;
+  inline TransactionId getTransactionId() const { return mTransactionId; }
 
+protected:
   /**
-   * Called in freezeWith(Client) just before the transaction body is built. The
-   * intent is for the derived class to assign their data variant to the
-   * transaction body.
+   * Derived from Executable. Construct a transaction protobuf object from this transaction.
+   *
+   * @return The transaction protobuf object that contains this transaction information.
    */
-  virtual void onFreeze(proto::TransactionBody* body) const = 0;
+  virtual proto::Transaction makeRequest() const = 0;
```

Comment:
Do these pure virtual functions need to be declared here? They are already pure virtual in `Executable`, so won't all child classes to this one be forced to implement regardless?

Reply:
You're right they don't, got copy-paste happy it seems. Removed from here and in `Query` as well.